spec/BLSMinimalPublicKeySize.cry

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2020 Galois, Inc.
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+*/
+
+module BLSMinimalSignatureSize = BLSGeneric where
+
+// Suite BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_NUL_, as defined in
+// draft-irtf-cfrg-bls-signature-04, Secton 4.2.1 and Appendix A
+
+import Parameters
+import HashToCurveE2
+import Pairing (bls_ate_pairing)
+import Serialization
+import ShortWeierstrassCurve as EC
+
+// a pairing-friendly elliptic curve
+type t_F = t_Fp_2 // representation type for field of the primary curve (signature)
+type t_F' = t_Fp // representation type for field of the secondary curve (pubkey)
+type t_G = t_Fp_12  // representation type for field of the pairing result
+
+type pubkey_len = 48*8
+type signature_len  = 96*8
+
+main_curve = E'
+other_curve = E
+
+// hash_to_point  BLS12381G1_XMD:SHA-256_SSWU_RO_
+type DST_len = 43
+
+DST: [DST_len][8]
+DST = "BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_NUL_"
+
+hash_to_point hx = hash_to_curve (DST, hx)
+
+// other primitives "determined by the above parameters"
+
+P: EC::AffinePoint t_F' // base point
+P = BP
+
+// pairing: (EC::AffinePoint t_F, EC::AffinePoint t_F') -> t_G
+pairing (p,q) = bls_ate_pairing p q // Section 2.2
```

Comment:
I think `p` and `q` need to be swapped here in the call to `bls_ate_pairing`, right?  Otherwise the sizes are wrong and this causes a type error.